### DISCUSSION TITLE
SDK interface to non FHIR server systems

### DISCUSSION THREAD

**Discussion by dennisbrox** [FirelyTeam/firely-net-sdk#1962] [General]
SDK capitalizes class properties in a C# appropriate way that is not FHIR standard.  Going to a FHIR server through a client is not a problem but writing a resource class directly to Mongodb or the AWS FHIR api is.  How do you get around that?  Or is there a Python type FHIR library for C# from someone that does work?

**Comment by mbaltus** [comment]
If you use the Fhir[Json/Xml]Serializer, you can serialize your POCO's to correct FHIR json/xml. This is how FHIR data is usually communicated between two systems. If the receiving system cannot work with that, you will have to implement extra code to align with their API, or you would have to create an extra layer on the receiving end to convert the FHIR data to the format it can work with.

**Comment by dennisbrox** [comment]
I am not communicating between 2 systems.  I'm using the create and update
methods that take resources and which would seem the obvious way to do it.
I'd rather pass them serialized data so I could see what they are doing and
force the server to get what I wanted to send it.

**Reply by mbaltus** [reply]
Create and Update are the FhirClient methods that are used to send FHIR data to a FHIR server - your side acting as the client. So that would in my view be communicating between two systems.

**Reply by mbaltus** [reply]
The FhirClient does send out the serialized data by the way. So it should be possible to inspect that on the HTTP call.

**Comment by dennisbrox** [comment]
It's encrypted on the way I presume so seeing it seems impossible and I
actually know what's in it anyway.  It's the AWS Lambda service that is not
handling it properly and it's virtually impossible to modify that without a
fleet of programmers.

**Reply by mbaltus** [reply]
Sounds like AWS still has some work to do before they can call themselves FHIR compliant. I know you've tried to reach out to their support, but maybe you can file tickets on their github repos to try and get this improved.

**Comment by dennisbrox** [comment]
Interesting.  Their "open source" FHIR server is what they call a Lambda
function.  A user can create a Lambda function their handle whatever
service function they want .   The FHIR server is generated automatically
when you set up an account and it has some configuration parameters you can
set but nothing related to actual functionality.  It is a node js program
that is very large.  If you don't like what it does, you can change it
yourself if you can hire a fleet of programmers to do it.  Tech support I
buy does not know anything about it and do not support it in any way.
Obviously they have someone who put it together who knows what they are
doing but they are more inaccessible than God.  I  may go back to Mongo DB
and deal with the problem that SDK classes are not FHIR compatible, same
problem that Cosmos DB seems to have, and find a way to store serialized
data.

**Reply by Brilbroeder** [reply]
found a solution to store fhir-resource from the sdk in mongodb ? mongodb is complaining about types not being configured and allowed to be serialized
